File: Examples/VisualBasic/ProgrammersGuide/Working-with-Pages/ExportOfHiddenVisioPagesToHTML.vb
Imports VisualBasic
Imports Aspose.Diagram
Imports System
Imports Aspose.Diagram.Saving

Public Class ExportOfHiddenVisioPagesToHTML
    Public Shared Sub Run()
        'ExStart:ExportOfHiddenVisioPagesToHTML
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Intro()

        ' load an existing Visio
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))
        ' get a particular page
        Dim page As Page = diagram.Pages.GetPage("Flow 2")
        ' set Visio page visiblity
        page.PageSheet.PageProps.UIVisibility.Value = BOOL.[False]

        ' initialize PDF save options
        Dim options As New HTMLSaveOptions()
        ' set export option of hidden Visio pages
        options.ExportHiddenPage = False

        'Save the Visio diagram
        diagram.Save(dataDir & Convert.ToString("ExportOfHiddenVisioPagesToHTML_Out.html"), options)
        'ExEnd:ExportOfHiddenVisioPagesToHTML
    End Sub
End Class

File: Examples/CSharp/Working-with-Pages/ExportOfHiddenVisioPagesToHTML.cs
﻿using Aspose.Diagram;
using Aspose.Diagram.Saving;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_with_Pages
{
    public class ExportOfHiddenVisioPagesToHTML
    {
        public static void Run()
        {
            try
            {
                // ExStart:ExportOfHiddenVisioPagesToHTML
                // The path to the documents directory.
                string dataDir = RunExamples.GetDataDir_Intro();

                // Load an existing Visio
                Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");
                // Get a particular page
                Page page = diagram.Pages.GetPage("Flow 2");
                // Set Visio page visiblity
                page.PageSheet.PageProps.UIVisibility.Value = UIVisibilityValue.Visible;

                // Initialize PDF save options
                HTMLSaveOptions options = new HTMLSaveOptions();
                // Set export option of hidden Visio pages
                options.ExportHiddenPage = false;
                // Set export option of comments
                options.IsExportComments = false;

                // Save the Visio diagram
                diagram.Save(dataDir + "ExportOfHiddenVisioPagesToHTML_out.html", options);
                // ExEnd:ExportOfHiddenVisioPagesToHTML
            }
            catch (System.Exception ex)
            {
                System.Console.WriteLine(ex.Message + "\nThis example will only work if you apply a valid Aspose License. You can purchase full license or get 30 day temporary license from http:// Www.aspose.com/purchase/default.aspx.");
            }
        }
    }
}
